### ISSUE TITLE
usage of asset_filter breaks target

### ISSUE THREAD

**Issue by leapfog** [zyedidia/eget#64]
With the following simple config file
```
[global]
  target = "~/bin"
  upgrade_only = true
```
the command `eget neovim/neovim` downloads the `appimage` and places it in `~/bin`.

But when I add
```
["neovim/neovim"]
   asset_filters = "nvim.appimage"
```
the download ends up in `./` (i.e. the current directory).

**Comment by patinthehat** [comment]
It looks like this might be a bug related to the expansion of the tilde (~). Can you verify that it works correctly if you use an absolute pathname in the `[global]` section for `target`?

i.e.,

```toml
[global]
  target = "/home/someuser/bin"
  upgrade_only = true
```

**Comment by leapfog** [comment]
No, that doesn't help.

But testing that, it seems, that adding `asset_filters = "nvim.appimage"` simply (and silently!) breaks reading the global section.

**Comment by patinthehat** [comment]
@leapfog Does using the array syntax fix this? Just trying to narrow down the possibilities here.

```toml
["neovim/neovim"]
   asset_filters = [ "nvim.appimage" ]
```

**Comment by hhromic** [comment]
I think this is an effect of what I just reported in #84.
When a configuration file is malformed, `eget` simply ignores it and uses defaults if no other valid config files are found.
The default for `target` is to save the binaries in the current directory.

**Comment by zyedidia** [comment]
I believe this is fixed by #85.
